lib/ansible/modules/remote_management/manageiq/manageiq_tenant.py

```diff
@@ -0,0 +1,447 @@
+#
+# (c) 2018, Evert Mulder (base on manageiq_user.py by Daniel Korn <korndaniel1@gmail.com>)
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+
+module: manageiq_tenant
+
+short_description: Management of tenants in ManageIQ.
+extends_documentation_fragment: manageiq
+version_added: '2.7'
+author: Evert Mulder
+description:
+  - The manageiq_tenant module supports adding, updating and deleting tenants in ManageIQ.
+
+options:
+  state:
+    description:
+    - absent - tenant should not exist, present - tenant should be.
+    choices: ['absent', 'present']
+    default: 'present'
+  name:
+    description:
+      - The tenant name.
+    required: true
+    default: null
+  description:
+    description:
+    - The tenant description.
+    required: true
+    default: null
+  parent_id:
+    description:
+    - The id of the parent tenant. If not supplied the root tenant is used.
+    required: false
+    default: null
+  quotas:
+    description:
+    - The tenant quotas.
+    - All parameters case sensitive.
+    - 'Valid attributes are:'
+    - ' - C(cpu_allocated) (int): use null to remove the quota.'
+    - ' - C(mem_allocated) (GB): use null to remove the quota.'
+    - ' - C(storage_allocated) (GB): use null to remove the quota.'
+    - ' - C(vms_allocated) (int): use null to remove the quota.'
+    - ' - C(templates_allocated) (int): use null to remove the quota.'
+    required: false
+    default: null
+'''
+
+EXAMPLES = '''
+- name: Update the root tenant in ManageIQ
+  manageiq_tenant:
+    name: 'My Company'
+    description: 'My company name'
+    manageiq_connection:
+      url: 'http://127.0.0.1:3000'
+      username: 'admin'
+      password: 'smartvm'
+      verify_ssl: False
+
+- name: Create a tenant in ManageIQ
+  manageiq_tenant:
+    name: 'Dep1'
+    description: 'Manufacturing department'
+    parent_id: 1
+    manageiq_connection:
+      url: 'http://127.0.0.1:3000'
+      username: 'admin'
+      password: 'smartvm'
+      verify_ssl: False
+
+- name: Delete a tenant in ManageIQ
+  manageiq_tenant:
+    state: 'absent'
+    name: 'Dep1'
+    parent_id: 1
```

Comment:
I assume parent_id not necessary for deletion?

Reply:
Ah :+1:.  I should have started by saying I don't know anything about tenants in manageiq :)